core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -169,36 +169,46 @@ public void onNext(HttpObject o) {
                     break;
                 }
 
+                if (req.method() == HttpMethod.HEAD) {
+                    // HEAD responses always close the stream with the initial headers, even if not explicitly
+                    // set.
+                    endOfStream = true;
+                } else {
+                    final int statusCode = status.code();
+                    switch (statusCode) {
+                        case 204:
+                        case 205:
+                        case 304:
```

Comment:
Curious if we can reuse `ArmeriaHttpUtil.isContentAlwaysEmpty(int)`. It has a check for informational class statuses, though.

Reply:
Good idea I think it makes sense - using the helper now. Also touched the helper a little for better readability / less duplication.